wondrous-app/components/Common/CSVFileDropZone/index.tsx

```diff
@@ -0,0 +1,74 @@
+import { Grid } from '@mui/material';
+import { useCSVReader, formatFileSize } from 'react-papaparse';
+import palette from 'theme/palette';
+import { DropZone, DropZoneFileContainer, DropZoneFileName, DropZoneFileRemoveButton, DropZoneIcon } from './styles';
+
+interface ICSVFileDropzoneProps {
+  handleFileUpload: (results) => void;
+  handleFileRemove?: () => void;
+  dropText?: string;
+  isDisabled?: boolean;
+}
+
+const CSVFileDropzone = (props: ICSVFileDropzoneProps) => {
+  const {
+    handleFileUpload,
+    handleFileRemove,
+    dropText = 'Drop CSV file here or click to upload',
+    isDisabled = false,
+  } = props;
+  const { CSVReader } = useCSVReader();
+
+  return (
+    <CSVReader
+      onUploadAccepted={(results: any) => {
+        handleFileUpload(results?.data);
+      }}
+      noDrag={isDisabled}
+      noClick={isDisabled}
+      skipEmptyLines
+    >
+      {({ getRootProps, acceptedFile, getRemoveFileProps, Remove }: any) => (
```

Comment:
Do not use ‘any’ keyword when you know what type of data your variable’s gonna hold. It is recommended that you always define the data type whenever you declare a new variable.